src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js

```diff
@@ -164,49 +173,55 @@ const ActiveState = ({
-                                    onReplyEdit={onReplyUpdateHandler(item.id)}
-                                    onReplySelect={onCommentSelectHandler(item.id)}
-                                    onShowReplies={onShowRepliesHandler(item.id, item.type)}
-                                    repliesTotalCount={item.total_reply_count}
-                                    replies={item.replies}
-                                    translations={translations}
-                                >
-                                    {hasNewThreadedReplies ? (
-                                        <BaseComment
-                                            {...item}
-                                            currentUser={currentUser}
-                                            getAvatarUrl={getAvatarUrl}
-                                            getMentionWithQuery={getMentionWithQuery}
-                                            getUserProfileUrl={getUserProfileUrl}
-                                            mentionSelectorContacts={mentionSelectorContacts}
-                                            onDelete={onCommentDelete}
-                                            onEdit={onCommentEdit}
-                                            onSelect={onCommentSelectHandler(item.id)}
-                                            permissions={{
-                                                can_delete: getProp(item.permissions, 'can_delete', false),
-                                                can_edit: getProp(item.permissions, 'can_edit', false),
-                                                can_reply: getProp(item.permissions, 'can_reply', false),
-                                                can_resolve: getProp(item.permissions, 'can_resolve', false),
-                                            }}
-                                            translations={translations}
+                                {hasNewThreadedReplies ? (
+                                    <BaseComment
+                                        {...item}
+                                        {...commentProps}
+                                        onDelete={onCommentDelete}
+                                        onEdit={onCommentEdit}
+                                        onSelect={onCommentSelectHandler(item.id)}
+                                        permissions={{
+                                            can_delete: getProp(item.permissions, 'can_delete', false),
+                                            can_edit: getProp(item.permissions, 'can_edit', false),
+                                            can_reply: getProp(item.permissions, 'can_reply', false),
+                                            can_resolve: getProp(item.permissions, 'can_resolve', false),
+                                        }}
+                                    >
+                                        <Replies
```

Comment:
Pros are that we can have an isolation of props going to Replies instead of feeding all of the reply specific props to BaseComment. Theres some callbacks so there would also be one less level to drill. Another pro is that if we want replies to other activity in the future, we have this readable pattern we can follow in ActiveState. 

Cons are that we have duplication of props in `commentProps`. ActiveState seems a bit bulky as well. There's also concerns on the usage of children in this case. If we want Replies to modify the parentComment it would be harder to do as well.

I didn't realize the intention in the RFC was to have the replies in the Comment. I think Jared had some concerns on the usage of children for Replies as well, so I think I can change it

Reply:
Agreed offline to put Replies in the BaseComment. We added Replies to the BaseComment file to avoid the circular dependency warning.